src/rnr_score_hv/harvester_base.py

```diff
@@ -0,0 +1,89 @@
+"""
+Copyright 2022 NOAA
+All rights reserved.
+
+Collection of methods to facilitate file/object retrieval
+
+"""
+import argparse
+
+from rnr_score_hv.yaml_utils import YamlLoader
+from rnr_score_hv import harvesters as hvs
+
+def harvest(harvest_config):
+    """
+    Gets harvester config as either a yaml file or dict and returns
+    a list of tuples
+
+    Parameters
+    ----------
+    harvest_config: str
+        The dict or yaml file containing the harvester configuration
+
+    Returns
+    -------
+    requested_data: list
+        A list of tuples where each tuple's contents is defined by the
+        harvester
+    """
+
+    # Convert incoming config (either dictionary or file) to dictionary
+    if isinstance(harvest_config, dict):
+        harvest_dict = harvest_config
+    else:
+        # Create dictionary from the input file
+        harvest_dict = YamlLoader(harvest_config).load()[0]
+
+    # Get the list of applications
+    try:
```

Comment:
can you add a few lines of comments of what is happening in this block of code? I also see that even the base class requires some specific fields in the config. Should that be documented in the harvest() header?